tools/cli/src/mentat_cli/repl.rs

```diff
@@ -200,51 +232,99 @@ impl Repl {
+        let mut start = PreciseTime::now();
+        let mut end: Option<PreciseTime> = None;
 
         match cmd {
-            Command::Help(args) => self.help_command(args),
-            Command::Timer(on) => self.toggle_timer(on),
-            Command::Cache(attr, direction) => self.cache(attr, direction),
+            Command::Cache(attr, direction) => {
+                self.cache(attr, direction);
+            },
+            Command::Close => {
+                self.close();
+            },
+            Command::Exit => {
+                self.close();
+                eprintln!("Exiting…");
+                process::exit(0);
+            },
+            Command::Help(args) => {
+                self.help_command(args);
+            },
+            Command::Import(path) => {
+                self.execute_import(path);
+            },
             Command::Open(db) => {
                 match self.open(db) {
                     Ok(_) => println!("Database {:?} opened", self.db_name()),
                     Err(e) => eprintln!("{}", e.to_string()),
                 };
             },
-            Command::Import(path) => self.execute_import(path),
-            Command::Sync(args) => {
-                match self.store.sync(&args[0], &args[1]) {
-                    Ok(_) => println!("Synced!"),
-                    Err(e) => eprintln!("{:?}", e)
-                };
-            }
             Command::OpenEmpty(db) => {
                 match self.open_empty(db) {
                     Ok(_) => println!("Empty database {:?} opened", self.db_name()),
                     Err(e) => eprintln!("{}", e.to_string()),
                 };
             },
-            Command::Close => self.close(),
-            Command::Query(query) => self.execute_query(query),
-            Command::QueryExplain(query) => self.explain_query(query),
+            Command::Query(query) => {
+                self.store
+                    .q_once(query.as_str(), None)
+                    .map_err(|e| e.into())
+                    .and_then(|o| {
+                        end = Some(PreciseTime::now());
+                        self.print_results(o)
+                    })
+                    .map_err(|err| {
+                        eprintln!("{:?}.", err);
+                    })
+                    .ok();
+            },
+            Command::QueryExplain(query) => {
+                self.explain_query(query);
+            },
+            Command::QueryPrepared(query) => {
+                self.store
+                    .q_prepare(query.as_str(), None)
+                    .and_then(|mut p| {
+                        let prepare_end = PreciseTime::now();
+                        if should_time {
+                            eprint_out("Prepare time");
+                            eprint!(": ");
+                            format_time(start.to(prepare_end));
+                        }
+                        // This is a hack.
```

Comment:
Right, because timing assumes 1-to-1 mapping between command & "thing to be timed", whereas here it's a 1-to-many.